### PR TITLE
sw_engine : arm neonRasterGrayscale8 function to support aarch64

### PR DESCRIPTION
**Pull request by rinechran**
Improved the speed through neon processing

### CODE REVIEW

**Comment by rinechran** [comment]
https://github.com/thorvg/thorvg/issues/30

**Comment by hermet** [comment]
@rinechran I really appreciate your contribution. Could you please share the performance before/after? I'd like to suggest you to capture the report values at sampling number [1000] among "Lottie" / "Performance" examples.

**Comment by rinechran** [comment]
Files ending with 'aarch64' are newly added log files. 'aarch32' files operate on armv7. Additionally, files with an underscore '_' indicate the speed before the patch.
```
Lottie_aarch64  : [  934]: update = 0.059118s,   raster = 0.025865s,  total = 0.084983s
Lottie_aarch32(new)   : [  934]: update = 0.059891s,   raster = 0.025821s,  total = 0.085713s
Lottie_ :  [  468]: update = 0.058781s,   raster = 0.026280s,  total = 0.085061s

Performance_aarch64.txt : [  999]: rendering = 0.013876s,  average = 0.014811s
Performance_aarch32.txt(new) : [ 1000]: rendering = 0.014088s,  average = 0.014918s
Performance_ : [ 1000]: rendering = 0.014276s,  average = 0.015231s
```

[Lottie_aarch32.txt](https://github.com/thorvg/thorvg/files/15196285/Lottie_aarch32.txt)
[Lottie_aarch64.txt](https://github.com/thorvg/thorvg/files/15196287/Lottie_aarch64.txt)
[Performance_aarch32.txt](https://github.com/thorvg/thorvg/files/15196288/Performance_aarch32.txt)
[Performance_aarch64.txt](https://github.com/thorvg/thorvg/files/15196289/Performance_aarch64.txt)
[Lottie_.txt](https://github.com/thorvg/thorvg/files/15196308/Lottie_.txt)
[Performance_.txt](https://github.com/thorvg/thorvg/files/15196309/Performance_.txt)

**Comment by rinechran** [comment]
Files ending with 'aarch64' are newly added log files. 'files with an underscore '_' indicate the speed before the patch.
```
Lottie_aarch64  : [  934]: update = 0.059118s,   raster = 0.025865s,  total = 0.084983s
Lottie_ :  [  468]: update = 0.058781s,   raster = 0.026280s,  total = 0.085061s

Performance_aarch64.txt : [  999]: rendering = 0.013876s,  average = 0.014811s
Performance_ : [ 1000]: rendering = 0.014276s,  average = 0.015231s
```

[Lottie_aarch64.txt](https://github.com/thorvg/thorvg/files/15196287/Lottie_aarch64.txt)
[Performance_aarch64.txt](https://github.com/thorvg/thorvg/files/15196289/Performance_aarch64.txt)
[Lottie_.txt](https://github.com/thorvg/thorvg/files/15196308/Lottie_.txt)
[Performance_.txt](https://github.com/thorvg/thorvg/files/15196309/Performance_.txt)

**Comment by hermet** [comment]
Improvements Rate:
- Lottie: (0.025986/0.026201) = +4.7%
- Performance: (0.014163/0.014785) = +4%

**Comment by hermet** [comment]
@rinechran thanks, please double check your commit msg: I don't see this is a good form as a change history.
image

You might wanna leave meaningful information there like:
```
Improvements Rate:
Lottie: (0.025986/0.026201) = +4.7%
Performance: (0.014163/0.014785) = +4.3%
```

**Comment by rinechran** [comment]
Yes, I got it
I was looking for a way to improve the performance, and it's history. Let me summarize

File: src/renderer/sw_engine/tvgSwRasterNeon.h [hunk 1]
```diff
@@ -24,6 +24,16 @@
 
 #include <arm_neon.h>
 
+//TODO : need to support windoiws ARM
```
Comment (hermet) [inline src/renderer/sw_engine/tvgSwRasterNeon.h:27]: typo:  windoiws

File: src/renderer/sw_engine/tvgSwRasterNeon.h [hunk 2]
```diff
@@ -24,6 +24,16 @@
...
+ 
+//https://developer.arm.com/documentation/101028/0012/13--Advanced-SIMD--Neon--intrinsics
```
Comment (hermet) [inline src/renderer/sw_engine/tvgSwRasterNeon.h:29]: I don’t see the need for this link. Let's leave only necessary comments. 
People can easily search for them as needed.

**Review by hermet — APPROVED** [review]

**Pull request merged**